### PR TITLE
opensles fixes

### PR DESCRIPTION
**Pull request by zavadovsky**
Fix mem leak and noised sound playing on Android.

### CODE REVIEW

File: channels/rdpsnd/client/opensles/rdpsnd_opensles.c [hunk 1]
```diff
@@ -136,32 +136,7 @@ static void rdpsnd_opensles_set_format(rdpsndDevicePlugin* device,
...
-						break;
-				}
-				break;
-
-			case WAVE_FORMAT_ADPCM:
-			case WAVE_FORMAT_DVI_ADPCM:
-				opensles->format = format->wFormatTag;
-				break;
-		}
-
```
Comment (akallabeth) [inline channels/rdpsnd/client/opensles/rdpsnd_opensles.c]: @zavadovsky Why did you remove this?

**Comment by zavadovsky** [comment]
@akallabeth 
With this lines client is doing this:
rdpsnd get ADPCM -> decode as ADPCM -> get raw PCM -> play sound -> all OK
rdpsnd get PCM -> send raw PCM data to ADPCM decoder -> get garbage -> play noise

Without this lines:
rdpsnd get ADPCM -> decode as ADPCM -> get raw PCM -> play sound -> all OK
rdpsnd get raw PCM -> play sound -> all OK

Tested on WinSrv2008 and Win7 - they send ADPCM.
And on latest Xrdp - it sends raw PCM.

**Comment by bmiklautz** [comment]
@zavadovsky @zavadovsky maybe this depends on the android device you are testing on?

**Comment by zavadovsky** [comment]
@bmiklautz I test on Xperia Z3 and Galaxy Note 2014.
Client from playmarket - play noise on both.
Myself builded from "1.2.0-beta1+android9" tag - play noise on both.
With this fix - play normal sound.
As i said before - as server i'm using Xrdp from its master branch on Debian Wheezy.

**Comment by bmiklautz** [comment]
@akallabeth you remember what tests (device and server) we did?

**Comment by akallabeth** [comment]
@bmiklautz Did some tests against win7, win8 and freerdp. Just compiled and run the pull, looks good.

**Comment by zavadovsky** [comment]
@bmiklautz @akallabeth 
You may try to apply this patch and connect Android client to Windows and play some sound:
```diff
diff --git a/channels/rdpsnd/client/rdpsnd_main.c b/channels/rdpsnd/client/rdpsnd_main.c
index b7e9eae..96dd652 100644
--- a/channels/rdpsnd/client/rdpsnd_main.c
+++ b/channels/rdpsnd/client/rdpsnd_main.c
@@ -169,6 +169,9 @@ void rdpsnd_select_supported_audio_formats(rdpsndPlugin* rdpsnd)
 	{
 		serverFormat = &rdpsnd->ServerFormats[index];
 
+		if (serverFormat->wFormatTag != WAVE_FORMAT_PCM)
+			continue;
+
 		if (rdpsnd->fixedFormat > 0 && (rdpsnd->fixedFormat != serverFormat->wFormatTag))
 			continue;
 
```
This patch disables all audio formats except PCM.
I make this test without my fix and get same noise from Win7 as i got from Xrdp.
With my fix - sound is normal.

**Comment by zavadovsky** [comment]
Hello.

@bmiklautz @akallabeth @awakecoding 
ping

I rebased this commits to actual master branch.

**Comment by akallabeth** [comment]
LGTM, +1
@awakecoding @hardening @bmiklautz Looks like this one has been open for quite some time, could we merge?
